src/lib/y2network/hwinfo.rb

```diff
@@ -113,6 +113,8 @@ def hwinfo_from_hardware(name)
       #
       # @return [HardWrapper]
       def hardware_wrapper
+        return @hardware_wrapper if @hardware_wrapper
+
         @hardware_wrapper = HardwareWrapper.new
```

Comment:
It might be more concise:

```ruby
@hardware_wrapper ||= HardwareWrapper.new
```

However, it is true that being so "concise" brought us to this situation :-)